src/storage/mod.rs

```diff
@@ -511,34 +503,15 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
     /// The prewrite phase of a transaction. The first phase of 2PC.
     ///
     /// Schedules a [`CommandKind::Prewrite`].
-    pub fn prewrite(
-        &self,
-        ctx: Context,
-        mutations: Vec<Mutation>,
-        primary: Vec<u8>,
-        start_ts: TimeStamp,
-        lock_ttl: u64,
-        skip_constraint_check: bool,
-        txn_size: u64,
-        min_commit_ts: TimeStamp,
-        callback: Callback<Vec<Result<()>>>,
-    ) -> Result<()> {
-        check_key_size!(
-            mutations.iter().map(|m| m.key().as_encoded()),
-            self.max_key_size,
-            callback
-        );
+    pub fn prewrite(&self, cmd: Command, callback: Callback<Vec<Result<()>>>) -> Result<()> {
+        if let CommandKind::Prewrite(commands::Prewrite { mutations, .. }) = &cmd.kind {
```

Comment:
The cmd is required as Prewrite type, can we use a strong constraint to guarantee it here?

Reply:
This is addressed in a follow-up PR.